apps/address-book/app/components/App/App.js

```diff
@@ -40,21 +41,26 @@ const App = () => {
 
   return (
     <Main assetsUrl={ASSETS_URL}>
-      <Header
-        primary="Address Book"
-        secondary={
-          <Button mode="strong" icon={<IconPlus />} onClick={newEntity} label="New Entity" />
-        }
-      />
+      { entries.length === 0 && <Empty action={newEntity} /> }
+      { entries.length > 0 &&
```

Comment:
Do we need to render `<Main>` and `<IdentityProvider>`, no matter what? 

If not, an early return would read more clearly, in my opinion:

```js
if (!entries.length) return <Empty action={newEntity} />

return (
  <Main>
    ...
```

If we do need to render them, then maybe we could do something like this:

```js
// at the top level of the file
const Wrap = ({ children }) => {
  const { api } = useAragonApi()

  const handleResolveLocalIdentity = address =>
    api.resolveAddressIdentity(address).toPromise()

  const handleShowLocalIdentityModal = address =>
     api.requestAddressIdentityModification(address).toPromise()

  return (
    <Main  assetsUrl={ASSETS_URL}>
      <IdentityProvider
        onResolve={handleResolveLocalIdentity}
        onShowLocalIdentityModal={handleShowLocalIdentityModal}
      >
        {children}
      </IdentytProvider>
    </Main>
  )
}

// in App component
if (!entries.length) return <Wrap><Empty action={newEntity} /></Wrap>

return (
  <Wrap>
    <Header ... />
    <Entities ... />
    <SidePanel ... />
  </Wrap>
)

Reply:
Oh, I just saw your comment about always needing `SidePanel`. It's easy to rework my suggestion to put `SidePanel` in `Wrap`.

Reply:
This is my code so far:

    const Wrap = ({ children }) => (
    <Main assetsUrl={ASSETS_URL}>
      { children }
      <IdentityProvider
        onResolve={handleResolveLocalIdentity}
        onShowLocalIdentityModal={handleShowLocalIdentityModal}
      >
        { entries.length > 0 &&
          <Entities
            entities={entries}
            onNewEntity={newEntity}
            onRemoveEntity={removeEntity}
          />
        }
        <SidePanel onClose={closePanel} opened={panelVisible} title="New entity">
          <NewEntity onCreateEntity={createEntity} />
        </SidePanel>
      </IdentityProvider>
    </Main>
    )

    if (!entries.length) return (
    <Wrap><Empty action={newEntity} /></Wrap>
    )

    return (
    <Wrap>
      <Header
        primary="Address Book"
        secondary={
          <Button mode="strong" icon={<IconPlus />} onClick={newEntity} label="New Entity" />
        }
      />
    </Wrap>
    )

But I'm not sure how to get rid of that last inline check. Also, I'm not sure if this solution really contributes to clarity.